qiskit/ignis/characterization/fitters.py

```diff
@@ -18,37 +18,47 @@
 Fitters of characteristic times
 """
 
+from typing import Union, List, Callable, Optional, Tuple, Dict
 from scipy.optimize import curve_fit
 import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib import axes
```

Comment:
These 2 lines can't happen at the module level it is what is causing the CI failure, matplotlib is not a requirement for ignis (or any of ignis's dependencIes) and is optional. So we need to make this import happen at run time if a visualization function is called.